python/pyarrow/tests/test_parquet.py

```diff
@@ -1024,6 +1024,24 @@ def test_read_partitioned_directory(tmpdir):
     _partition_test_for_filesystem(fs, base_path)
 
 
+@parquet
+def test_create_parquet_manifest_with_thread_pool(tmpdir):
+    fs = LocalFileSystem.get_instance()
+    base_path = str(tmpdir)
+
+    _partition_test_for_filesystem(fs, base_path)
+
+    import pyarrow.parquet as pq
+    from concurrent.futures import ThreadPoolExecutor
+
+    thread_pool_executor = ThreadPoolExecutor(8)
+    dataset = pq.ParquetManifest(base_path, filesystem=fs,
+                                 thread_pool=thread_pool_executor)
```

Comment:
Add extra `ParquetDataset` using here per above. Maybe check that the number of partitions is the same in the threaded vs. unthreaded cases?